### ISSUE TITLE
Some magnet links work, others remain inactive

### ISSUE THREAD

**Issue by frankvw2017** [rakshasa/rtorrent#850]
With increasing frequency I've noted that some magnet links work while others just sit there indefinitely without one bit being downloaded; they create a .meta file but nothing else. I have been unable to find any indication as to the difference between magnet links that work and ones that don't.

Unfortunately at this point in time the only links I can site that illustrate the problem refer to illegal content (i.e. copyrighted material) so the following are provided ONLY as a troubleshooting tool and are NOT to be taken as any endorsement of copyright infringement. (Suggestions on how to illustrate this problem without referring to illegal content would be appreciated!)

The following works:
`magnet:?xt=urn:btih:2ef439b382be776a2fba840e5617302b4b086deb&dn=Westworld.S02.Complete.720p.WEB-DL.x264.%5BMP4%5D.%5B5.3GB%5D.%5BSeason.2%5D&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Ftracker.openbittorrent.com%3A80&tr=udp%3A%2F%2Fopen.demonii.com%3A1337&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Fexodus.desync.com%3A6969`

The following does not work:
`magnet:?xt=urn:btih:b55d936f9beedd74df7b86c774243ded0ff337ab&dn=Minions.Yellow+is+the+New+Black.2019.1080p.Bluray.X264-EVO%5BTGx%5D&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Ftracker.openbittorrent.com%3A80&tr=udp%3A%2F%2Fopen.demonii.com%3A1337&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Fexodus.desync.com%3A6969`

Also, the .torrent file in the session directory generated from the above two magnet links differ. The working magnet link produces a torrent file that will work in the Transmission bittorrent client while the one the doesn't work produces a corrupt torrent file that won't workHowever, the magnet link that does cause problems in rtorrent does work in Transmission. If I convert the offending magnet link to a .torrent file (using) magnet2torrent.com or a similar online converter and then load this torrent file into rtorrent, the download starts right away.

Contents or .rtorrent.rc:

```
# rtorrent rc file

# Maximum number of concurrent uploads per torrent:
max_uploads = 5

# Global upload and download rate in KiB; "0" for unlimited:
download_rate = 0
upload_rate = 10

# Default directory to save the downloaded torrents:
directory = /home/me/bittorrent/incoming

# Session directory for meta files:
session = /home/me/bittorrent/session

# Watch a directory for new torrents, and stop those that have been
# deleted
schedule = watch_directory,5,5,load_start=/home/me/bittorrent/torrentfiles/*.torrent

# Prevent certificate issues:
network.http.ssl_verify_peer.set=0

# Accept encrypted connections as an option:
encryption =  require,require_RC4,allow_incoming,try_outgoing

# Enable DHT:
dht = auto
dht_port = 6881
peer_exchange = yes

#EOF
```

`
What can I do to further troubleshoot this  problem?

**Comment by chros73** [comment]
Easiest workaround is to [use aria2 to get the torrent file](https://github.com/chros73/rtorrent-ps-ch_setup/blob/master/ubuntu-14.04/home/chros73/bin/addMagnetLinksAria2.sh) from the magnet link.

**Comment by frankvw2017** [comment]
Yes, converting  magnet links to torrent files does work as I wrote in my OP. But that sort of defeats the purpose of rTorrent supporting magnet links in the place. The issue is that some magnet links do work while others just sit there (a problem that does not occur with the torrent files created from those magnet links) which means there is an issue with rTorrent's magnet link support that deserves to be reported.

**Comment by mhertz** [comment]
Little late sorry, but just wanted to say that the link you state doesn't work, works fine at my end on rtorrent-ps-ch, and I just tested latest stable vanilla rtorrent working also. Anyway, I believe it's a known issue that you cannot force encryption with rtorrent when using magnets, so if relaxing your settings for this, then it will work again, and I also tested with your setting which also didn't work then.

Note, encryption in a torrent-client are only good for overcoming traffic-shaping/throttling, and not usefull otherwise, as the persons you need hiding from is the companies monitoring the swarms and contacts your ISP with your IP, and not the ISP itself, who doesn't actively monitor packets for such.

**Comment by rakshasa** [comment]
Both UDP tracker issues and DHT magnet links have been fixed.
